### ISSUE TITLE
Component Accessibility Status: Enforce playwright tag naming convention

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#15370]
We've established that all e2e tests ran with playwright should have one of four tags:

https://github.com/carbon-design-system/carbon/blob/670dffcae241d9bf9b3a64ea06e100e7e5a08c37/docs/testing.md?plain=1#L46-L53

I think we can use `eslint-plugin-playwright` to enforce this naming convention now that they've added a `valid-title` rule in https://github.com/playwright-community/eslint-plugin-playwright/issues/167

```[tasklist]
- [x] Add `eslint-plugin-playwright`
- [x] Ensure it's only ran against the proper files, like `e2e/components/accordion/accordion-test.e2e.js`, and not the "e2e" test files for icons, etc that use jest and not playwright
- [x] Configure `valid-title` to ensure playwright tests with the `-test.avt.e2e.js` filename suffix titles always have one of the 4 allowable tags (5 if you count the blanket `@avt` tag we also use in the `describe` blocks.
- [x] Without a valid title, eslint should error (not warn)
- [x] Add some tests in the PR that show it failing on a test without a proper tag. These can be removed after the PR is approved and before it's merged.
- [ ] Fix any remaining errors that pop up from this
```
